XamarinCommunityToolkit/Views/CameraView/CameraViewRenderer.uwp.cs

```diff
@@ -0,0 +1,399 @@
+			var file = await destinationFolder.CreateFileAsync($"{DateTime.Now.ToString("yyyyddMM_HHmmss")}.mp4", CreationCollisionOption.GenerateUniqueName);
+			filePath = file.Path;
+			if (Element.VideoStabilization)
+			{
+				var stabilizerDefinition = new VideoStabilizationEffectDefinition();
+				videoStabilizationEffect = (VideoStabilizationEffect)await mediaCapture.AddVideoEffectAsync(stabilizerDefinition, MediaStreamType.VideoRecord);
+				var recommendation = videoStabilizationEffect.GetRecommendedStreamConfiguration(mediaCapture.VideoDeviceController, encodingProfile.Video);
+				if (recommendation.InputProperties != null)
+				{
+					inputPropertiesBackup = mediaCapture.VideoDeviceController.GetMediaStreamProperties(MediaStreamType.VideoRecord) as VideoEncodingProperties;
+					await mediaCapture.VideoDeviceController.SetMediaStreamPropertiesAsync(MediaStreamType.VideoRecord, recommendation.InputProperties);
+					await mediaCapture.VideoDeviceController.SetMediaStreamPropertiesAsync(MediaStreamType.VideoPreview, recommendation.InputProperties);
+				}
+				if (recommendation.OutputProperties != null)
+				{
+					outputPropertiesBackup = encodingProfile.Video;
+					encodingProfile.Video = recommendation.OutputProperties;
+				}
+				videoStabilizationEffect.Enabled = true;
+			}
+			mediaRecording = await mediaCapture.PrepareLowLagRecordToStorageFileAsync(encodingProfile, file);
+			await mediaRecording.StartAsync();
+		}
+
+		async Task<MediaSource> StopRecord()
+		{
+			if (mediaRecording == null)
+				return null;
+
+			await mediaRecording.StopAsync();
+			await mediaRecording.FinishAsync();
+			mediaRecording = null;
+
+			if (videoStabilizationEffect != null)
+			{
+				await mediaCapture.RemoveEffectAsync(videoStabilizationEffect);
+				videoStabilizationEffect = null;
+
+				if (inputPropertiesBackup != null)
+				{
+					await mediaCapture.VideoDeviceController.SetMediaStreamPropertiesAsync(MediaStreamType.VideoRecord, inputPropertiesBackup);
+					inputPropertiesBackup = null;
+				}
+				if (outputPropertiesBackup != null)
+				{
+					encodingProfile.Video = outputPropertiesBackup;
+					outputPropertiesBackup = null;
+				}
+				videoStabilizationEffect = null;
+			}
+
+			return MediaSource.FromFile(filePath);
+		}
+
+		void MediaCaptureFailed(MediaCapture sender, MediaCaptureFailedEventArgs errorEventArgs)
+			=> Element?.RaiseMediaCaptureFailed(errorEventArgs.Message);
+
+		protected override async void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			switch (e.PropertyName)
+			{
+				case nameof(CameraView.CameraOptions):
+					await CleanupCameraAsync();
+					await InitializeCameraAsync();
+					break;
+				case nameof(CameraView.FlashMode):
+					if (flash != null)
+						flash.IsEnabled = Element.FlashMode == CameraFlashMode.Torch || Element.FlashMode == CameraFlashMode.On;
+					break;
+				case nameof(CameraView.PreviewAspect):
```

Comment:
Do we will support this? I did a quick look and I didn't find it on iOS, but we have it on android.
cc: @jfversluis

Reply:
As indicated [here](https://github.com/xamarin/XamarinCommunityToolkit/issues/211) (by the original author) indeed this is only for UWP. Strange that this is not a platform specific then while the rest is. I'll keep it for now, but might be worth to take it out and keep this is something platform specific to not confuse users.

Reply:
I commented it out and left a note that it's only supported by Android and we can revisit when we have the platform specifics.  In the original spec, there is also MirrorPreview that is only android, but I didn't see any code in CameraView.

@pictos You might want to look into the MirrorPreview on the Android renderer to see if it's needed https://github.com/xamarin/Xamarin.Forms/blob/ae045fe259a23984ea5790b6551fc1e3fb3f5159/Xamarin.Forms.Platform.Android/Renderers/CameraViewRenderer.cs#L74-L76

Reply:
@bmacombe, yes on android this method is used with PlatformSpecific, I did a spike and you find it [here](https://github.com/jfversluis/XamarinCommunityToolkit-1/commit/faf782d8d28193c3f3764505022fda1ce9ffde3c). I'm just awaiting the feedback from the team to see how we will handle it. But if until my PR we don't have a decision we can remove it and add later as enchantment.

Reply:
@pictos Cool sounds good!

Reply:
@pictos if you're comfortable bring up that spike link in the forms channel on Discord. If you'd rather have me do it, let me know :)

Reply:
@jfversluis I can do it (: